extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -91,7 +91,6 @@ public QueryDefinition makeQueryDefinition(
     ShuffleSpec nextShuffleSpec = findShuffleSpecForNextWindow(operatorList.get(0), maxWorkerCount);
```

Comment:
isn't choosing `MixShuffleSpec.instance()` a strategy choice for the current set of operators?
is there a special reason which warrants to build the query and then rebuild it again? can't we have these changes inside the `findShuffleSpecForNextWindow` method?

Reply:
The problem is that we don't want the MixShuffleSpec to be propagated in the query context to all the previous GroupBy/Scan stages, as that would be inefficient. Hence the changes of this PR are different than `findShuffleSpecForNextWindow` (as we only want the MixShuffleSpec in the just-previous-stage, not in all the previous GroupBy/Scan stages)

Reply:
it seems to me that its pretty linear from here to where you've built the plan and replaced the last shuffle with something else...why would a change inside this method would alter all those?
the shuffle you have overriden there I guess is the one which was returned here - isn't that true?

Reply:
`findShuffleSpecForNextWindow()` returns null when we have empty over clause.

In my original commit on this PR, I changed the logic of `findShuffleSpecForNextWindow()` to be:
```java
if (partition == null) {
      return null;
    }
if (partition.getPartitionColumns().isEmpty()) {
      return MixShuffleSpec.instance();
    }
```

But Adarsh pointed out that this would propagate MixShuffleSpec to the other query kits which is inefficient.

For example, it would make `shuffleSpec=MixShuffleSpec` for queries where we have multiple Scan stages before the Window stage. Sample query: 
```sql
WITH "ext" AS (
  select * from wikipedia left join drill_wikipedia on wikipedia."countryName" = drill_wikipedia."countryName" where wikipedia."countryName" in ('Guatemala') limit 5
)
SELECT
  "__time",
  "countryName",
  "channel",
  row_number() over (order by countryName, cityName, channel) as c1,
  count(channel) over (partition by cityName order by countryName, cityName, channel) as c2
FROM "ext"
group by __time, countryName, cityName, channel 
order by countryName, cityName, channel
```

The above query gives:
![image](https://github.com/user-attachments/assets/fcf2211b-2b2c-4d57-8526-78a5e75a7f53)


Hence, it's better to not propagate MixShuffleSpec to the other query kits, and just override it here only for the just-previous stage.

Hope that makes sense!

Reply:
how the current changes in the PR differ from writing:
```
    ShuffleSpec nextShuffleSpec =  nextShuffleSpec isEmptyOverPresent ?  MixShuffleSpec.instance() : findShuffleSpecForNextWindow(operatorList.get(0), maxWorkerCount);
```
at the line this comment was placed at originally?

Reply:
There's a difference in what gets passed in the query context to GroupBy/Scan query kits.

In the PR code changes, `null` gets passed, indicating the query kits that they can keep shuffling of the previous stage.

In the suggested code change, we would pass `MixShuffleSpec.instance().clusterBy()` == `ClusterBy(Collections.emptyList(), 0)` which is different than null.